sdk/core/azure-core/azure/core/_pipeline_client_async.py

```diff
@@ -48,9 +50,8 @@
     SensitiveHeaderCleanupPolicy,
 )
 
-
-HTTPRequestType = TypeVar("HTTPRequestType")
-AsyncHTTPResponseType = TypeVar("AsyncHTTPResponseType", bound="AsyncContextManager")
+AsyncHTTPResponseType = TypeVar("AsyncHTTPResponseType", AsyncHttpResponse, LegacyAsyncHttpResponse)
+HTTPRequestType = TypeVar("HTTPRequestType", HttpRequest, LegacyHttpRequest)
```

Comment:
I disagree with this strongly, this file is type agnostic on purpose, PipelineClient works for any type, is not tight to our implementation